docs/howto/writing-migrations.txt

```diff
@@ -184,6 +184,45 @@ the respective field according to your needs.
   migration is running. Objects created after the ``AddField`` and before
   ``RunPython`` will have their original ``uuid``’s overwritten.
 
+.. _non-atomic-migrations:
+
+Non-atomic migrations
+~~~~~~~~~~~~~~~~~~~~~
+
+On databases that support DDL transactions (SQLite and PostgreSQL), migrations
+will be run inside a transaction. For some use-cases, like performing data
+migrations on large tables, you may want to exempt a migration from being run
+in a transaction by setting the ``atomic`` class attribute to ``False``::
+
+    from django.db import migrations
+
+    class Migration(migrations.Migration):
+
+        atomic = False
+
+``RunPython`` operations will still be run inside a transaction unless
```

Comment:
I think we should make the relationship between migrations and operations a bit more clear e.g.: "Within such a migration, individual `RunPython` operations will still be run inside a transaction. Other operations are run without a transaction."

Could you clarify why you made `RunPython` default to `atomic=True` rather than to the value of `atomic` in its containing migration? I think both are valid designs. I'd like to make sure we think this decision through.

With the current proposal, a very long `RunPython` operation declared inside a non-atomic migration could tie the database with a long-runnnig transaction and cause downtime. Per the principle of least astonishment, I'm slightly in favor of inheriting the migration's value of `atomic`.

There are probably other concerns that I didn't think about, though.